### PR TITLE
Static Analysis corrections on c++ model classes

### PR DESCRIPTION
**Pull request by datumbox**
Minor refactoring based on static analysis and manual review of the code of models:
- Eliminate unnecessary value parameter to constant reference to avoid unnecessary copies.
- Eliminate unused include.

Post discussion with @fmassa, we might not want these changes to keep the C++ API easy to understand. Let me know if there is any thing we want to cherrypick from here else feel free to close the PR.

### CODE REVIEW

File: torchvision/csrc/models/densenet.h [hunk 1]
```diff
@@ -23,10 +23,10 @@ struct VISION_API DenseNetImpl : torch::nn::Module {
   torch::nn::Sequential features{nullptr};
   torch::nn::Linear classifier{nullptr};
 
-  DenseNetImpl(
+  explicit DenseNetImpl(
```
Comment (fmassa) [inline torchvision/csrc/models/densenet.h:26]: for my understanding: is this the default recommended way of constructing models in C++?

Note that we do not use `explicit` in our other model implementations.

cc @anjali411 do you have any recommendation here?

**Review by fmassa — COMMENTED** [review]
Changes look good to me overall, but I have a question wrt `explicit` constructors.

Also, FYI I believe we have disabled C++ model tests in torchvision CI https://github.com/pytorch/vision/blob/d8ccbe7ec11e11400e608fd1f53d6e48badce55e/setup.py#L169 so this might not be tested on CI.

File: torchvision/csrc/models/densenet.h (hunk 1 above)
Comment (datumbox) [reply torchvision/csrc/models/densenet.h]: @fmassa Thanks for flagging it. As discussed on call, it's probably better to drop this and revisit the static analysis recommendation when I'm more certain about the implications of this. :)

**Comment by datumbox** [comment]
> so this might not be tested on CI.

Thanks for letting me know. I re-enabled locally the C++ tests on `test_cpp_models.py` and they passed. The only ones failing on macOS are the MNASNet which I did not update in this PR. So we should be good.

**Review by fmassa — APPROVED** [review]
Thanks!

**Pull request merged**